omega/omega-connector/omega-connector-grpc/src/main/java/org/apache/servicecomb/pack/omega/connector/grpc/core/ReconnectStreamObserver.java

```diff
@@ -42,10 +45,23 @@ public ReconnectStreamObserver(
   public void onError(Throwable t) {
     LOG.error("Failed to process grpc coordinate command.", t);
     loadContext.getGrpcOnErrorHandler().handle(messageSender);
+    cancelWait();
   }
 
   @Override
   public void onCompleted() {
     // Do nothing here
   }
+
+  public void cancelWait(){
+    latch.countDown();
+  }
+
+  public void waitConnected() {
```

Comment:
When will you call the waitConnected?
I don't think it's a good practice to introduce sync wait here.

Reply:
> When will you call the waitConnected?

Waiting only happens when Omega first connects to Alpha. 
I marked the location where call waitConnected in GrpcSagaClientMessageSender.java L77

> I don't think it's a good practice to introduce sync wait here.

The original idea was when Alpha receives the connection request, it sends a response message to Omega. Omega receives the response message and then continues.

if you don't feel it's needed  I would remove them

Reply:
done